### PR TITLE
[dygraph] Add faster_rcnn_enhance

### PR DESCRIPTION
**Pull request by liuhuiCNN**
(1) Add faster_rcnn_enhance, on coco dataset mAP=41.9 (41.6-->41.9), on coco_mini mAP=35.7 (34.9-->35.7).

### CODE REVIEW

File: dygraph/configs/faster_rcnn_enhance/_base_/faster_fpn_reader.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+worker_num: 2
+TrainReader:
+  sample_transforms:
+  - DecodeOp: {}
+  - RandomFlipOp: {prob: 0.5}
+  - AutoAugmentOp: {autoaug_type: v1}
+  - NormalizeImageOp: {is_scale: true, mean: [0.485,0.456,0.406], std: [0.229, 0.224,0.225]}
+  - ResizeImage: {target_size: [384, 416, 448, 480, 512, 544, 576, 608, 640, 672], max_size: 1000, interp: 1, use_cv2: true}
```
Comment (yghstill) [inline dygraph/configs/faster_rcnn_enhance/_base_/faster_fpn_reader.yml:8]: 把ResizeImage 换成 RandomResizeOp，refer to solov3 3x config.

File: dygraph/ppdet/modeling/losses/diou_loss.py [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+        cy = (y1 + y2) / 2
+        w = x2 - x1
+        h = y2 - y1
+
+        cxg = (x1g + x2g) / 2
+        cyg = (y1g + y2g) / 2
+        wg = x2g - x1g
+        hg = y2g - y1g
+
+        x2 = fluid.layers.elementwise_max(x1, x2)
```
Comment (yghstill) [inline dygraph/ppdet/modeling/losses/diou_loss.py:71]: 把所有fluid API都要换成paddle的。

File: dygraph/ppdet/modeling/losses/giou_loss.py [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        self.num_classes = num_classes
+        self.do_average = do_average
+        self.class_weight = 2 if is_cls_agnostic else num_classes
+        self.use_class_weight = use_class_weight
+
+    # deltas: NxMx4
+    def bbox_transform(self, deltas, weights):
+        wx, wy, ww, wh = weights
+
+        deltas = fluid.layers.reshape(deltas, shape=(0, -1, 4))
```
Comment (yghstill) [inline dygraph/ppdet/modeling/losses/giou_loss.py:59]: same as above.

Comment (liuhuiCNN) [reply dygraph/ppdet/modeling/losses/giou_loss.py]: done

File: dygraph/ppdet/modeling/losses/diou_loss.py (hunk 2 above)
Comment (liuhuiCNN) [reply dygraph/ppdet/modeling/losses/diou_loss.py]: done

File: dygraph/configs/faster_rcnn_enhance/_base_/faster_fpn_reader.yml (hunk 1 above)
Comment (liuhuiCNN) [reply dygraph/configs/faster_rcnn_enhance/_base_/faster_fpn_reader.yml]: done

**Pull request closed without merging**